extensions/amp-app-banner/0.1/amp-app-banner.js

```diff
@@ -36,6 +34,12 @@ export class AbstractAppBanner extends AMP.BaseElement {
   constructor(element) {
     super(element);
 
+    /**
+     * @return {!../../../src/service/url-impl.Url}
+     * @protected
+     */
+    this.getUrlService = once(() => Services.urlForDoc(this.element));
```

Comment:
AFAIK `services` handles caching of result already, don't see a need for `once`

Reply:
Will remove all usages of once() for `Url` service in a follow up PR.

Reply:
(I did it originally for lazy-instantiation, but given that the URL service is instantiated initially there's no need for it).